client/packages/programs/src/JsonForms/components/AdherenceStatus.tsx

```diff
@@ -0,0 +1,172 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import {
+  composePaths,
+  ControlProps,
+  rankWith,
+  uiTypeIs,
+} from '@jsonforms/core';
+import { withJsonFormsControlProps } from '@jsonforms/react';
+import { BasicTextInput, Box } from '@openmsupply-client/common';
+import {
+  FORM_INPUT_COLUMN_WIDTH,
+  FORM_LABEL_COLUMN_WIDTH,
+  useZodOptionsValidation,
+} from '../common';
+import { z } from 'zod';
+import { useEncounter } from '../../api';
+import { get as extractProperty } from 'lodash';
+import { FormLabel } from '@mui/material';
+
+export const adherenceStatusTester = rankWith(10, uiTypeIs('AdherenceStatus'));
+
+type Options = {
+  /**
+   * Field name for the dispense count is stored.
+   * This field is loaded from the previous encounter.
+   *
+   * For example: `previousDispensedCountField: 'arvMedication.numberOfDaysDispensed'`
+   */
+  previousCountField: string;
+  /**
+   * Field name of the remaining count field.
+   * For example: `remainingCountField: 'arvMedication.remainingPillCount'`
+   */
+  remainingCountField: string;
+  /** Expected number of pills per day that a patient is suppose to take. */
+  countPerDay: number;
+
+  /** Location where to store the adherence status */
+  targetField: string;
+};
+
+const Options: z.ZodType<Options> = z
+  .object({
+    previousCountField: z.string(),
+    remainingCountField: z.string(),
+    countPerDay: z.number(),
+    targetField: z.string(),
+  })
+  .strict();
+
+const UIComponent = (props: ControlProps) => {
+  const { data, handleChange, label, uischema, path } = props;
+  const [targetPath, setTargetPath] = useState<string | undefined>();
+  const [adherenceStatus, setAdherenceStatus] = useState<number | undefined>();
+
+  const { errors, options } = useZodOptionsValidation(
+    Options,
+    uischema.options
+  );
+
+  useEffect(() => {
+    if (!options) {
+      return;
+    }
+    const targetPath = composePaths(path, options.targetField);
+    setTargetPath(targetPath);
+    setAdherenceStatus(extractProperty(data, targetPath));
+  }, [options, path]);
+  useEffect(() => {}, [targetPath]);
+
+  // fetch current encounter
+  const encounterId = useEncounter.utils.idFromUrl();
+  const { data: currentEncounter, mutate: fetchEncounter } =
```

Comment:
Good catch. The mutate is used to fetch the encounter right now without caching . I used it because this was how the hook was already implemented. In a stacked PR https://github.com/openmsupply/open-msupply/pull/934 I already changed it to use `useQuery` so that the encounter isn't queried multiple times to render the same view...